Cosminis/DataAccess/FriendsRepo.cs

```diff
@@ -0,0 +1,457 @@
+using DataAccess.Entities;
+using CustomExceptions;
+using Models;
+using System.Data.SqlClient;
+using DataAccess;
+using Microsoft.EntityFrameworkCore;
+using System.Linq;
+
+namespace DataAccess;
+ 
+public class FriendsRepo : IFriendsDAO
+{
+    private readonly wearelosingsteamContext _context;
+    private readonly IUserDAO _userRepo;
+
+    public FriendsRepo(wearelosingsteamContext context, IUserDAO userRepo)
+    {
+        _context = context;
+        _userRepo = userRepo;
+    }
+
+    public List<Friends> GetAllRelationships()
+    {
+        return _context.Friends.ToList();
+    }
+
+    public List<Friends> ViewAllFriends(int userIdToLookup)
+    {
+        User userToLookup = _context.Users.Find(userIdToLookup);
+
+        IEnumerable<Friends> friendsQuery =
+            (from Friends in _context.Friends
+            where (Friends.UserIdTo == userToLookup.UserId) || (Friends.UserIdFrom == userToLookup.UserId)
+            select Friends);
+
+        if(friendsQuery == null)
+        {
+            throw new ResourceNotFound("This is one lonely user. At this point they should simply quit social media.");
+        }
+
+        try
+        {       
+            List<Friends> friendsList = friendsQuery.ToList();
+            
+            if(friendsList.Count() < 1)
+            {
+                throw new ResourceNotFound();
+            }
+
+            return friendsList;
+        }
+        catch(Exception E)
+        {
+            throw;
+        }                                  
+    }
+    
+    public Friends FriendsByUserIds(int searchingUserId, int user2BeSearchedFor)
```

Comment:
I have zero clue what you are doing in this method in general, but since you are claiming it works, all I can say is "yass queen" in caps